src/concat.js

```diff
@@ -13,8 +13,8 @@ var invoker = require('./invoker');
  * @category List
  * @sig [a] -> [a] -> [a]
  * @sig String -> String -> String
- * @param {Array|String} a
- * @param {Array|String} b
+ * @param {Array|String|Object} a
+ * @param {Array|String|Object} b
```

Comment:
What about the second argument that can be an object if used with a dispatching first argument?

Like in this test:

```js
var z1 = { x: 'z1', concat: function(that) { return this.x + ' ' + that.x; } };
var z2 = { x: 'z2' };
eq(R.concat(z1, z2), 'z1 z2');
```